Azure/communication-ui-library
# bugfix: Do not show stale errors when a user rejoins a call

# What

Before this PR, stale errors were sometimes shown in composites via error bar when a user rejoins a call.

| Before | After |
| --- | --- |
| ![Before](https://user-images.githubusercontent.com/82062616/203393385-4fc28781-076d-4862-ac47-71a8769161bf.gif) | ![After](https://user-images.githubusercontent.com/82062616/203393394-dcc1cdf7-bafb-41bd-bb4b-281ab5df3517.gif) |

Repro steps:

- Join a group call using `CallComposite` or `CallWithChatComposite`
- Make sure your camera is in use by another application
- Try turning on video
  - You will see an error bar saying "failed to start video". This is expected.
- Leave the call
- Rejoin call from the end call screen.
  - The "failed to start video" error bar will reappear. *This is a bug*.

This PR adds a prop to the `ErrorBar` component, `ignorePremountErrors`. When set, the component does not show errors that occurred before the component was mounted.

The composites set this prop so that errors from previous calls are ignored.

# How Tested

- New unit-tests
- Make sure bug is fixed with the repro steps (see screen-captures above)
- Make sure error bar is _not_ dismissed when opening people / chat pane in mobile view (because the primary call pane is hidden in that case)
- sanity test hold and lobby screens
